File: boj/Gold/1034. 램프/램프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;

public class Main {
	
	static int n;
	static int m;
	static String[] s;
	static Map<String, Integer> map;

    public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	StringTokenizer st = new StringTokenizer(br.readLine());
    	
    	n = Integer.parseInt(st.nextToken());
    	m = Integer.parseInt(st.nextToken());
    	
    	map = new HashMap<>();
    	int ans = 0;
    	s = new String[n];
    	for (int i = 0; i < n; i++) {
    		s[i] = br.readLine();
    	}
    	
    	int k = Integer.parseInt(br.readLine());
    	
    	for (int i = 0; i < n; i++) {
    		String str = s[i];
    		if (map.containsKey(str)) {
    			map.put(str, map.get(str) + 1);
    			ans = Math.max(ans, map.get(str));
    			continue;
    		}
    		
    		int res = 0;
    		for (int j = 0; j < m; j++) {
    			if (str.charAt(j) == '0') res++;
    		}
    		
    		if (res > k || (k - res) % 2 != 0) continue;
    		
    		map.put(str, 1);
    		ans = Math.max(ans, 1);
    	}
    	
    	System.out.println(ans);
    }
}

File: boj/Gold/1034. 램프/램프.cs
using System;
class Program
{
    static int n;
    static int m;
    static string[] input;
    static Dictionary<string, int> map;
    static void Main(string[] args)
    {
        string[] inp = Console.ReadLine().Split();
        n = int.Parse(inp[0]);
        m = int.Parse(inp[1]);

        input = new string[n];
        map = new Dictionary<string, int>();

        for (int i = 0; i < n; i++)
        {
            input[i] = Console.ReadLine();
        }

        int k = int.Parse(Console.ReadLine());

        int ans = 0;

        for (int i = 0; i < n; i++)
        {
            if (map.ContainsKey(input[i]))
            {
                map[input[i]]++;
                ans = Math.Max(ans, map[input[i]]);
                continue;
            }

            int res = input[i].Count(c => c == '0');

            if (res > k || (k - res) % 2 == 1) continue;

            ans = Math.Max(ans, 1);
            map[input[i]] = 1;
        }

        Console.Write(ans);
    }
}